server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1415,6 +1445,40 @@ public static Query unsignedLongRangeQuery(
             }
             return builder.apply(l, u);
         }
+
+        static PointInSetQuery bitmapIndexQuery(String field, RoaringBitmap bitmap) {
+            final BytesRef encoded = new BytesRef(new byte[Integer.BYTES]);
+            return new PointInSetQuery(field, 1, Integer.BYTES, new PointInSetQuery.Stream() {
```

Comment:
I'm a little bit concerned about the `PointInSetQuery`'s cost estimate.

1. It iterates through all of the points, which may be a lot of points.
2. If a point is contained in a leaf of the BKD tree, it assumes that half the points/docs in the leaf (256) match the query, which is likely way too high.

I just did a quick though experiment: Suppose we have 100M docs on a shard (for simplicity, assume the shard is a single segment). The BKD tree will have 100M/512 = 200K leaves. Assuming fairly uniform distribution of documents across the available ID space, I don't think it's out of the question for a query on 1M IDs to hit every one of those leaves. As a result, the cost estimate will be 50M (when the truth is likely less than 1M).

tl;dr: The PointInSetQuery provides a good cost estimate in the general case, but probably doesn't work well for the case where your points are essentially IDs.

We'll need to think about whether this is good enough. On the plus side, it doesn't affect how things are indexed, so we have the opportunity to make it better in the future.